tracer/src/Datadog.Trace/NativeLoader.cs

```diff
@@ -0,0 +1,66 @@
+// <copyright file="NativeLoader.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace
+{
+    internal static class NativeLoader
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(NativeLoader));
+
+        private static bool IsAvailable
+        {
+            get
+            {
+                var fd = FrameworkDescription.Instance;
+                return fd.ProcessArchitecture != ProcessArchitecture.Arm && fd.ProcessArchitecture != ProcessArchitecture.Arm64;
+            }
+        }
+
+        public static bool TryGetRuntimeIdFromNative(out string runtimeId)
+        {
+            if (!IsAvailable)
+            {
+                runtimeId = default;
+                return false;
+            }
+
+            try
+            {
+                var runtimeIdPtr = GetRuntimeId();
+                runtimeId = Marshal.PtrToStringAnsi(runtimeIdPtr);
+                return !string.IsNullOrWhiteSpace(runtimeId);
+            }
+            catch (Exception e)
+            {
+                // We failed to retrieve the runtime from native this can be because:
+                // - P/Invoke issue (unknown dll, unknown entrypoint...)
+                // - We are running in a partial trust environment
+                Log.Warning("Failed to get the runtime-id from native: {Reason}", e.Message);
```

Comment:
Given there are still legitimate cases where we will get this (e.g. Datadog.Trace.dll custom instrumentation only, no automatic), I've think we _should_ still keep this as an info?

Reply:
👍